### DISCUSSION TITLE
Str::ucwords Missing From Support File

### DISCUSSION THREAD

**Discussion by siscodev93** [laravel/framework#44268] [Ideas] [answered]
Hello Everyone! 

The Str support class located in the Illuminate\Support namespace provides numerous helpful functions for manipulating strings and making it easy to format them in any manner that suits your needs.  The support class has various functions for modifying the capitalization of characters in the string including :
- upper, 
- lower,
- studly, 
- snake, 
and even ucfirst.  However the class is missing the 'ucwords' function call for returning a string where  each words first letter becomrd capitalized.

I haven't created a PR for a framework before, and wanted to touch base to  see if there is a reason the function wasn't implemented or if it's proper to just create a branch and submit the PR?

Thanks Folk!

**Comment by juse-less** [comment, marked as answer]
I believe this is what [`Str::title()`](https://laravel.com/docs/9.x/helpers#method-title-case) does.

**Reply by shaedrich** [reply]
Maybe, there should be aliases because there are several such helpers with non-intuitive names or at least people might know that functionality under different, potentially more popular names like studly vs pascal vs upperCamel.

**Reply by shaedrich** [reply]
See https://github.com/laravel/framework/discussions/47044#discussioncomment-7034511

**Comment by variablesign** [comment]
If am not mistaken, `ucwords` is not the same as `Str:title()`.

Example:
`ucwords('HeLLo woRld!') => 'HeLLo WoRld!'`
`Str::title('HeLLo woRld!') => 'Hello World!'`

**Reply by shaedrich** [reply]
That is not really something, Laravel could easily solve by its current implementation. The complexity to respect all preferences and variations is something, we briefly touched on in #49572.

**Reply by variablesign** [reply]
> That is not really something, Laravel could easily solve by its current implementation. The complexity to respect all preferences and variations is something, we briefly touched on in #49572.

Okay, thanks.
